### PR TITLE
#263 Adds support for pi_heating_demand on Thermostats

### PR DESCRIPTION
**Pull request by michaelkrog**
This adds support for detecting heat demand on thermostats exposing pi_heating_demand.

If the feature is available, the OFF state will be added for current CurrentHeatingCoolingState and TargetHeatingCoolingState. Also a monitor will be setup to detect if the thermostat is heating or not.

A threshold (currently 10) is added so thermostat is not shown as heating if demand is below the threshold. This is aimed at making sure that radiators does not appear as heating when they are in fact still cold. (valve may open just a bit even when the room is 1-2 degrees warmer than target temperature)

### CODE REVIEW

**Comment by itavero** [comment]
Not sure if I have checked in the code formatting style, but I did notice that your code is formatted differently.

Also, you have added the target state "Off" as well, but what does it do if the user selects "Off" in the Home.app?

**Comment by michaelkrog** [comment]
I tried to run the lint target and it didnt complain about the formatting. I have tried to run a format on it now. 
Let me know if there is still some formating you would like different.

I am a bit unsure of the target OFF as well. If I do not add it, then it will always appear with a glowing icon in Homekit as if it is enabled. If I add it and it goes OFF, then icons stops glowing - but user will be able to toggle it on and off in the UI with no actual difference on the device.

So to sum it up for states relevant to a room where no heating occurs:
* CurrentState.OFF & TargetState.ON = Top left temperature indicator shows idle, device icon shows enabled.
* CurrentState.OFF & TargetState.OFF = Top left temperature indicator shows idle, device icon shows disabled.

**Comment by michaelkrog** [comment]
Thinking a bit more about it. 

The correct way must be to always have the TargetState=HEAT. That is the purpose of the device: Keep the heat at a specific degree. I'll revert that part.

**Comment by michaelkrog** [comment]
There.  Now it only touches CurrentState.

**Comment by michaelkrog** [comment]
This is no longer needed as running_state has been added to Danfoss Thermostats.
